frontend/src/devicelist/DevicesTable.jsx

```diff
@@ -20,132 +20,143 @@ import MaterialTable, { MTableToolbar } from "material-table";
+        title: "Actions",
+        field: "action",
+        filtering: false,
+        sorting: false,
+        render: function Actions(rowData) {
+          return <DeviceDetailsButton deviceInfo={rowData} />;
+        },
+        align: "center",
+        export: false
+      }
+    ];
+    this.detailPanel = [
+      {
+        icon: ExpandMore,
+        openIcon: ExpandLess,
+        tooltip: "Show Channels",
+        render: function DetailPanel(rowData) {
+          return (
+            <div className="lightestGrey" style={{ padding: "1.5em" }}>
+              <Typography variant="h6">Channels</Typography>
+              <ChannelDetailsTable channels={rowData.channels} />
+            </div>
+          );
+        }
+      }
+    ];
+    this.options = {
+      toolbar: true,
+      search: true,
+      exportButton: true,
+      headerStyle: {
+        backgroundColor: "#f1f1f1",
+        fontWeight: "bold"
+      },
+      filtering: true,
+      draggable: false
+    };
+    this.icons = {
+      Filter: FilterList,
+      Search,
+      ResetSearch: Clear,
+      SortArrow: ArrowDownward,
+      Export: SaveAlt,
+      FirstPage,
+      LastPage,
+      NextPage: ChevronRight,
+      PreviousPage: ChevronLeft
+    };
+  }
 
-function getOptions() {
-  return {
-    toolbar: true,
-    search: true,
-    exportButton: true,
-    headerStyle: {
-      backgroundColor: "#f1f1f1",
-      fontWeight: "bold"
-    },
-    filtering: true,
-    draggable: false
-  };
-}
+  getComponents() {
+    return this.components;
+  }
 
-function getIcons() {
-  return {
-    Filter: FilterList,
-    Search,
-    ResetSearch: Clear,
-    SortArrow: ArrowDownward,
-    Export: SaveAlt,
-    FirstPage,
-    LastPage,
-    NextPage: ChevronRight,
-    PreviousPage: ChevronLeft
-  };
-}
+  getColumnInfo() {
+    return this.columns;
+  }
+
+  getDetailPanel() {
+    return this.detailPanel;
+  }
+
+  getOptions() {
+    return this.options;
+  }
+
+  getIcons() {
+    return this.icons;
+  }
 
-export default function DevicesTable(props) {
-  const { title, devices } = props;
-  return (
-    <>
-      <Box>
-        <TableContainer style={{ maxHeight: 570 }}>
-          <MaterialTable
-            title={title}
-            components={getComponents()}
-            columns={getColumnInfo()}
-            data={devices}
-            detailPanel={getDetailPanel()}
-            options={getOptions()}
-            icons={getIcons()}
-          />
-        </TableContainer>
-      </Box>
-    </>
-  );
+  render() {
+    const { title, devices } = this.props;
+    return (
+      <>
+        <Box>
+          <TableContainer style={{ maxHeight: 570 }}>
+            <MaterialTable
+              title={title}
+              components={this.getComponents()}
+              columns={this.getColumnInfo()}
+              data={devices}
+              detailPanel={this.getDetailPanel()}
+              options={this.getOptions()}
+              icons={this.getIcons()}
```

Comment:
Can't we just pass `this.components` `this.columnInfo` `this.detailPanel` etc. rather than calling functions that just return those variables?